karapace/backup/api.py

```diff
@@ -390,14 +378,24 @@ def _handle_restore_topic(
         raise BackupTopicAlreadyExists(f"Topic to restore '{instruction.topic_name}' already exists")
 
 
+producer_exception: Exception | None = None
+
+
 def _raise_backup_error(exception: Exception) -> NoReturn:
-    raise BackupDataRestorationError("Error while producing restored messages") from exception
+    global producer_exception  # pylint: disable=global-statement
+    producer_exception = exception
```

Comment:
issue: Let's collect all exceptions that occur instead of just the last? I believe the `check_producer_exception()` call in `_handle_producer_send()` doesn't make any difference for the possibility of multiple underlying exceptions. In theory all `_handle_producer_send`s can be executed before the first actual send succeeds.